app/notification.go

```diff
@@ -212,14 +212,16 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 						return
 					}
 				}
+
+				updateFollowing := *a.Config().ServiceSettings.ThreadAutoFollow
 				if mentionType == ThreadMention || mentionType == CommentMention {
 					incrementMentions = false
+					updateFollowing = false
 				}
-
 				opts := store.ThreadMembershipOpts{
 					Following:             true,
```

Comment:
Not blocking but it feels a little weird to me that `Following` is true here even though it won't get updated to true since `UpdateFollowing` is false. Maybe it would make sense to change `Following` to a pointer and have it do the update any time it's not `nil` instead of having a separate `UpdateFollowing` field. Not pushing for any changes here, just something to think about for the future.

Reply:
That would remove a variable from the equation but wouldn't simplify the logic since it would result in encapsulating an additional state (being nil) into `Following`. I am thinking the additional `UpdateFollowing` field probably gives a bit more explicitness on the calling side and visibility on the underlying store logic. 

That said, I see how it can get confusing. I tend to think at `Following` as the state we are aiming to set and `UpdateFollowing` as the permission needed to do it so it makes sense to have the former `true` and the latter `false`.

Reply:
That's fair about the explicitness. It's just a little weird to me that if we just set `Following` it's a no-op if `UpdateFollowing` isn't set - so what's even the point of setting `Following`. That said, the pointer might be too implicit and as long as the whole thing is documented maybe it's safer to just keep it more explicit.

Reply:
I guess the way the code is currently written it allows for `UpdateFollowing` to be dynamic. Alternatively we could fill the `opts` object with only the needed fields for each case.

Reply:
`UpdateFollowing` is required when we don't want to change the following state . We can't rely on the zero value of  `ThreadMembershipOpts.Following` because it could set the following state to `false` (the zero value).

We can completely drop `UpdateFollowing` by doing a DB read `GetThreadMembershipForUser` and set the `Following` value based on the returned membership, but this would be inefficient.

To make the code simpler we could invert `UpdateFollowing` to `SkipUpdateFollowing`. This has two benefits
1. Makes the option even more explicit
2. Currently every time we want to change the following state we have to set two options `Following` and `UpdateFollowing`. By inverting `UpdateFollowing` to `SkipUpdateFollowing`, devs will only have to set the `Following` value if they want to change following state. In the rarer case of not needing a `Following` change, the dev can set `SkipUpdateFollowing`. On the other hand this might be "gotcha", because it sets the following state to `false` if both `Following` and `SkipUpdateFollowing` are omitted.

Reply:
Personally I am okay with the current approach. We can revisit this during a later refactor if needed.